server/game/player.js

```diff
@@ -682,7 +683,7 @@ class Player extends Spectator {
     }
 
     recyclePlots() {
-        if(this.plotDeck.isEmpty()) {
+        if(this.plotDeck.isEmpty() || (this.groupedPiles['plot deck'] && this.plotDeck.all(plot => plot.hasTrait('Scheme')))) {
```

Comment:
@ystros not amazingly happy with this, do you have a better idea?

Reply:
The rules accurate version of this would be to have a proper "is not considered in the plot deck" effect applied to the Schemes and then check how many cards *are* considered in the plot deck currently. If it's 0, then you recycle.

You could try to fake it w/ the current `cannotReveal` effect as we are currently, but I think that may cause Tricksy Crow to cause early recycles (assuming the effect doesn't wear off at the proper time). Worth looking at, perhaps.

Barring that, I'd honestly be more comfortable with checking that the agenda is Rains here rather than the presentational `groupedPiles` property. How things are grouped in the UI should not be affecting rules.